src/dev/build/tasks/patch_native_modules_task.js

```diff
@@ -16,62 +16,21 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+import install from '@elastic/simple-git/scripts/install';
+import { deleteAll } from '../lib';
+import path from 'path';
 
-import { scanCopy, untar, deleteAll } from '../lib';
-import { createWriteStream, mkdirSync } from 'fs';
-import { binaryInfo } from '../../../../x-pack/legacy/plugins/code/tasks/nodegit_info';
-import wreck from '@hapi/wreck';
-import { dirname, join, basename } from 'path';
-import { createPromiseFromStreams } from '../../../legacy/utils/streams';
-
-async function download(url, destination, log) {
-  const response = await wreck.request('GET', url);
-
-  if (response.statusCode !== 200) {
-    throw new Error(`Unexpected status code ${response.statusCode} when downloading ${url}`);
-  }
-  mkdirSync(dirname(destination), { recursive: true });
-  await createPromiseFromStreams([response, createWriteStream(destination)]);
-  log.debug('Downloaded ', url);
-}
-
-async function downloadAndExtractTarball(url, dest, log, retry) {
-  try {
-    await download(url, dest, log);
-    const extractDir = join(dirname(dest), basename(dest, '.tar.gz'));
-    await untar(dest, extractDir, {
-      strip: 1,
-    });
-    return extractDir;
-  } catch (e) {
-    if (retry > 0) {
-      await downloadAndExtractTarball(url, dest, log, retry - 1);
-    } else {
-      throw e;
-    }
-  }
-}
-
-async function patchNodeGit(config, log, build, platform) {
-  const plat = platform.isWindows() ? 'win32' : platform.getName();
-  const arch = platform.getNodeArch().split('-')[1];
-  const { downloadUrl, packageName } = binaryInfo(plat, arch);
-
-  const downloadPath = build.resolvePathForPlatform(platform, '.nodegit_binaries', packageName);
-  const extractDir = await downloadAndExtractTarball(downloadUrl, downloadPath, log, 3);
-
+async function patchGit(config, log, build, platform) {
```

Comment:
@spalger We replace nodegit with bundling native git binaries in Kibana for better performance, compatibility and stable. 
The lib is here https://github.com/elastic/simple-git 

Basically, for windows and mac, we download prebuilt binaries from the project https://github.com/desktop/dugite-native which is used by github desktop building it's git binaries.
For Linux, we modified some build scripts and build the binaries from the source to remove the dependency on libcurl and reduce the Glibc requirements to match the requirements of the kibana environment.

Reply:
Please review.